experiments/download_ssl4eo.py

```diff
@@ -0,0 +1,738 @@
+                )
+                cities.append(row)
+        return cities
+
+    @staticmethod
+    def get_interest_points(
+        cities: List[Dict[str, str]], size: int = 10000
+    ) -> List[List[float]]:
+        cities = sorted(cities, key=lambda c: int(c["population"]), reverse=True)[:size]
+        points = [[float(c["lng"]), float(c["lat"])] for c in cities]
+        return points
+
+    @staticmethod
+    def km2deg(kms: float, radius: float = 6371) -> float:
+        return kms / (2.0 * radius * np.pi / 360.0)
+
+    @staticmethod
+    def deg2km(deg: float, radius: float = 6371) -> float:
+        return deg * (2.0 * radius * np.pi / 360.0)
+
+
+class BoundedUniformSampler:
+    def __init__(self, boundaries: shape = None) -> None:
+        if boundaries is None:
+            self.boundaries = self.get_country_boundaries()
+        else:
+            self.boundaries = boundaries
+
+    def sample_point(self) -> List[float]:
+        minx, miny, maxx, maxy = self.boundaries.bounds
+        lon = np.random.uniform(minx, maxx)
+        lat = np.random.uniform(miny, maxy)
+        p = Point(lon, lat)
+        if self.boundaries.contains(p):
+            return [p.x, p.y]
+        else:
+            return self.sample_point()
+
+    @staticmethod
+    def get_country_boundaries(
+        download_root: str = os.path.expanduser("~/.cache/naturalearth"),
+    ) -> shape:
+        url = "https://www.naturalearthdata.com/http//www.naturalearthdata.com/download/110m/cultural/ne_110m_admin_0_countries.zip"  # noqa
+        filename = "ne_110m_admin_0_countries.shp"
+        if not os.path.exists(os.path.join(download_root, os.path.basename(url))):
+            download_and_extract_archive(url, download_root)
+        sf = shapefile.Reader(os.path.join(download_root, filename))
+        return shape(sf.shapes().__geo_interface__)
+
+
+class OverlapError(Exception):
+    pass
+
+
+def date2str(date: datetime) -> str:
+    return date.strftime("%Y-%m-%d")
+
+
+def get_period(date: datetime, days: int = 5) -> Tuple[str, str]:
+    date1 = date - timedelta(days=days / 2)
+    date2 = date + timedelta(days=days / 2)
+    return date2str(date1), date2str(date2)
+
+
+"""get collection and remove clouds from ee"""
+
+
+def maskS2clouds(image: ee.Image) -> ee.Image:
+    qa = image.select("QA60")
+    # Bits 10 and 11 are clouds and cirrus, respectively.
+    cloudBitMask = 1 << 10
+    cirrusBitMask = 1 << 11
+    # Both flags should be set to zero, indicating clear conditions.
+    mask = qa.bitwiseAnd(cloudBitMask).eq(0)
+    mask = mask.bitwiseAnd(cirrusBitMask).eq(0)
+    return image.updateMask(mask)
+
+
+def get_collection_s2a(cloud_pct: float = 20) -> ee.ImageCollection:
+    collection = ee.ImageCollection("COPERNICUS/S2_SR")
+    collection = collection.filter(ee.Filter.lt("CLOUDY_PIXEL_PERCENTAGE", cloud_pct))
+    # collection = collection.map(maskS2clouds)
```

Comment:
You can remove commented out code

Reply:
this one I thought it's good to keep as people also use this cloud masking commonly, I commented out for sake of time, but to be discussed.